reset_password.cgi

```diff
@@ -49,7 +48,14 @@ if ($cgi->param('do_save')) {
     if ($old_password eq $password_1) {
         ThrowUserError('new_password_same');
     }
-    validate_password($password_1, $password_2);
+
+    my $pwqc = Bugzilla->passwdqc;
+    if (not $pwqc->validate_password($password_1)) {
+        ThrowUserError("password_insecure", { reason => $pwqc->reason });
+    }
+    elsif ($password_1 ne $password_2) {
+        ThrowUserError("password_mismatch");
+    }
```

Comment:
you should be checking that `$password_1 eq $password_2` _before_ checking the password complexity.

this same logic is repeated multiple times, so either apply the same fix everywhere, or move `password_mismatch` checking into a common fn.